File: benchmark/tasks/queue-impl/verify.mjs
import { pathToFileURL } from "node:url";
import path from "node:path";
import assert from "node:assert/strict";

const workspace = process.argv[2];
if (workspace === undefined) {
    throw new Error("缺少 workspace 参数");
}

const { Queue } = await import(pathToFileURL(path.resolve(workspace, "src/queue.js")).href);

const queue = new Queue();
assert.equal(queue.size, 0);
assert.equal(queue.dequeue(), undefined);
assert.equal(queue.peek(), undefined);

queue.enqueue(1);
queue.enqueue(2);
queue.enqueue(3);
assert.equal(queue.size, 3);
assert.equal(queue.dequeue(), 1);
assert.equal(queue.dequeue(), 2);
assert.equal(queue.peek(), 3);
assert.equal(queue.size, 1);
assert.equal(queue.dequeue(), 3);
assert.equal(queue.dequeue(), undefined);

console.log("queue-impl 验收通过");

File: benchmark/tasks/queue-impl/verify.py
import importlib.util
import pathlib
import sys

workspace = pathlib.Path(sys.argv[1])
source = workspace / "src" / "queue.py"
spec = importlib.util.spec_from_file_location("queue_task", source)
assert spec is not None and spec.loader is not None, "无法加载 queue.py"
module = importlib.util.module_from_spec(spec)
sys.modules[spec.name] = module
spec.loader.exec_module(module)

queue = module.Queue()
assert queue.size == 0
assert queue.dequeue() is None
assert queue.peek() is None
queue.enqueue(1)
queue.enqueue(2)
queue.enqueue(3)
assert queue.size == 3
assert queue.dequeue() == 1
assert queue.dequeue() == 2
assert queue.peek() == 3
assert queue.size == 1
assert queue.dequeue() == 3
assert queue.dequeue() is None

print("queue-impl 验收通过")
